packages/handlersjs-logging/lib/loggers/pino-logger.ts

```diff
@@ -39,36 +39,69 @@ export class PinoLogger extends Logger {
 
     if (level <= this.minimumLevel) {
 
-      const logData = level > this.minimumLevelPrintData ? {} : data || {};
+      const logData = level > this.minimumLevelPrintData
+        ? {}
+        // if data is a string, convert it to an object with a data property. Pino expects data to be an object. If it remains a string, it will use the data as a message, and drop the rest of our log.
+        : typeof data === 'string'
+          ? { data }
+          // if data is undefined, convert it to an empty object.
+          : data ?? {};
+
       const logMessage = `[${this.label}] ${message}`;
 
-      // Pino does not use "silly", but "trace" instead. So we need to convert the level.
       const loggerOptions = {
-        level: level === 5 ? 'trace' : LoggerLevel[level].toString(),
+        // Define custom levels for the pino logger to match our own logger levels.
+        customLevels: {
+          error: 0,
+          warn: 1,
+          info: 2,
+          verbose: 3,
+          debug: 4,
+          silly: 5,
+        },
+        useOnlyCustomLevels: true,
+        level: LoggerLevel[level].toString(),
       };
 
-      const logger = this.prettyPrint ? Pino(loggerOptions, pretty()) : Pino(loggerOptions);
+      // if prettyPrint is ture, use pino-pretty with custom options to prettify the log. Otherwise, use pino without prettifying.
```

Comment:
```suggestion
      // if prettyPrint is true, use pino-pretty with custom options to prettify the log. Otherwise, use pino without prettifying.
```